x-pack/plugins/infra/public/plugin.ts

```diff
@@ -75,9 +66,10 @@ export class Plugin
       mount: async (params: AppMountParameters) => {
```

Comment:
is there a reason we don't just stick this whole function behind an async import?
```ts
mount: async (params: AppMountParameters) => {
  (await import('./apps/mount'))(params);
}
```

Reply:
I just think this is a lot easier to implement correctly and maintain

Reply:
Yeah, I will adopt this pattern in other plugins with less logic in the mount function. I'm not comfortable touching the plugin code here due to multiple dependencies between rendering functions.

Reply:
We agree and will clean up the mess for this plugin in a follow-up (#62413).